vscode-dotnet-runtime-library/src/Acquisition/GlobalInstallerResolver.ts

```diff
@@ -0,0 +1,303 @@
+        {
+            throw Error(`The requested version ${specificVersion} or resolved version is invalid.`);
+        }
+
+        const operatingSys : string = os.platform();
+        const operatingArch : string = os.arch();
+
+        let convertedOs = "";
+        let convertedArch = "";
+
+        switch(operatingSys)
+        {
+            case 'win32': {
+                convertedOs = 'win';
+                break;
+            }
+            case 'darwin': {
+                convertedOs = 'osx';
+                break;
+            }
+            case 'linux': {
+                convertedOs = operatingSys;
+                break;
+            }
+            default:
+            {
+                throw Error(`The OS ${operatingSys} is currently unsupported or unknown.`);
+            }
+        }
+
+        switch(operatingArch)
+        {
+            case 'x64': {
+                convertedArch = operatingArch;
+                break;
+            }
+            case 'x32': {
+                convertedArch = 'x86';
+                break;
+            }
+            case 'arm': {
+                convertedArch = operatingArch;
+                break;
+            }
+            case 'arm64': {
+                convertedArch = operatingArch;
+                break;
+            }
+            default:
+            {
+                throw Error(`The architecture ${operatingArch} is currently unsupported or unknown.`);
+            }
+        }
+
+        const desiredRidPackage = convertedOs + '-' + convertedArch;
+
+        const indexJson =  await this.fetchJsonObjectFromUrl(indexUrl);
+        const releases = indexJson['releases'];
+        if(releases.length == 0)
+        {
+            throw Error(`The releases json format used by ${indexUrl} is invalid or has changed, and the extension needs to be updated.`);
+        }
+
+        let sdks: any[] = [];
+        releases.forEach(function (release : any) {
+            sdks.push.apply(sdks, release['sdks']);
+        });
+
+        for (let sdk of sdks)
+        {
+            const thisSDKVersion : string = sdk['version'];
+            if(thisSDKVersion === specificVersion) // NOTE that this will not catch things like -preview or build number suffixed versions.
+            {
+               const thisSDKFiles = sdk['files'];
+               for (let installer of thisSDKFiles)
+               {
+                    if(installer['rid'] == desiredRidPackage && this.installerMatchesDesiredFileExtension(installer, convertedOs))
+                    {
+                        const installerUrl = installer['url'];
+                        if(installerUrl === undefined)
+                        {
+                            throw Error(`URL for ${desiredRidPackage} on ${specificVersion} is unavailable: The version may be Out of Support, or the releases json format used by ${indexUrl} may be invalid and the extension needs to be updated.`);
```

Comment:
This should be made into a specific event stream event, as well as the bottom 2 errors